src/omnicore/fees.cpp

```diff
@@ -43,6 +43,11 @@ void COmniFeeCache::UpdateDistributionThresholds()
         }
     }
 
+    // override above and use a fixed total count of 600,000 for OMNI & TOMNI
+    // this is done to remove the need for recalculating distribution thresholds every block (where dev OMNI fractionally changes the total number of tokens)
+    int64_t primaryTokenOverride = 600000 / OMNI_FEE_THRESHOLD;
```

Comment:
Updates are still done, it's just that UpdateDistributionThresholds() now takes a property ID and only updates the threshold for the property for the token count that was changed.

Currently we iterate all properties every time we call that function, which involves calling `getTotalTokens()` and thus an iteration of the tally map.  So that's hundreds of iterations of the tally map each time it's called.

By narrowing the scope down so we only update the threshold for the property where the token count has changed, `UpdateDistributionThresholds()` becomes a lot faster.

Hope that makes sense, when I'm back home I'll commit the proper code and hopefully it'll make sense :)